flink-queryable-state/flink-queryable-state-client-java/src/main/java/org/apache/flink/queryablestate/network/Client.java

```diff
@@ -419,20 +440,27 @@ void close() {
 		 * @param cause The cause to close the channel with.
 		 * @return Channel close future
 		 */
-		private boolean close(Throwable cause) {
+		private CompletableFuture<Boolean> close(Throwable cause) {
```

Comment:
I think this method can give you wrong results when being called twice. The second call will give you a completed future even though the first call could still be running.